newIDE/app/src/MainFrame/index.js

```diff
@@ -660,6 +663,40 @@ const MainFrame = (props: Props) => {
 
   useDiscordRichPresence(currentProject);
 
+  const serviceWorkerUpdateStatus = useServiceWorkerUpdateStatus();
+
+  useTimeout(
+    React.useCallback(
+      async () => {
+        if (serviceWorkerUpdateStatus === 'update-ready') {
```

Comment:
I would almost put this in a dedicated function in UpdateTools to avoid bloating too much MainFrame?

Reply:
Something like useServiceWorkerCheckAndAskToUpdate